zpay32/invoice.go

```diff
@@ -524,10 +513,12 @@ func validateInvoice(invoice *Invoice) error {
 		return fmt.Errorf("neither description nor description hash set")
 	}
 
-	// Can have at most 20 extra hops for routing.
-	if len(invoice.RoutingInfo) > 20 {
-		return fmt.Errorf("too many extra hops: %d",
-			len(invoice.RoutingInfo))
+	// Each route hint can have at most 20 hops.
+	for _, routeHint := range invoice.RouteHints {
```

Comment:
20 route hints should definitely be plenty when creating the invoice, but do we really want to mark an invoice as invalid just because it has more than that?

Reply:
We should at least restrict the number when creating invoices. I also think rejecting invoices with clearly excessive number of route hints (maybe this limit should be a bit higher than 20?) is reasonable.

Reply:
Fixed.